### PR TITLE
Sphrose/console options

### LINKED ISSUE #3973: [Feature] Additional toggle location for "Clear Console at game startup"
Currently the toggle is located in `Edit > Editor Settings > Global Preferences > General Settings > Clear Console at game startup`

image

I think it will be good if there is an additional toggle location that is more quick to access, like around one of these two spots:

image

### PR DESCRIPTION
**Pull request by sphrose**
## What does this PR do?

This PR resolves #3973. It adds a menu button to the right of the search bar in the console which provides an option to toggle the clear on play option that already exists in the settings. @rainbj I'd appreciate UX feedback on this.

## How was this PR tested?

Manual testing.

https://user-images.githubusercontent.com/82213493/210610349-34a00261-c77a-4da0-a197-7271a7e8d3ed.mp4

### CODE REVIEW

File: Code/Editor/Controls/ConsoleSCB.cpp [hunk 1]
```diff
@@ -296,6 +304,8 @@ void ConsoleLineEdit::ResetHistoryIndex()
 
 Lines CConsoleSCB::s_pendingLines;
 
+QMenu* optionsMenu;
+
```
Comment (cgalvan) [inline Code/Editor/Controls/ConsoleSCB.cpp:308]: I think this global variable was added on accident, please remove.

File: Code/Editor/Controls/ConsoleSCB.cpp [hunk 2]
```diff
@@ -384,6 +396,30 @@ CConsoleSCB::~CConsoleSCB()
...
+{
+    m_optionsMenu = new QMenu(QStringLiteral("Console Options Menu"), this);
+    connect(m_optionsMenu, &QMenu::aboutToShow, this, &CConsoleSCB::UpdateOptionsMenu);
+    ui->optionsButton->setMenu(m_optionsMenu);
+    ui->optionsButton->setAutoRaise(true);
+    ui->optionsButton->setPopupMode(QToolButton::InstantPopup);
+
+    m_clearOnPlayAction = new QAction(tr("Clear On Play"), this);
+    m_clearOnPlayAction->setCheckable(true);
+    connect(m_clearOnPlayAction, &QAction::triggered, this, [this] { ToggleClearOnPlay(); });
```
Comment (cgalvan) [inline Code/Editor/Controls/ConsoleSCB.cpp:409]: Nit: Can connect this straight to `ToggleClearOnPlay` if you make it a slot to avoid the lambda.

**Review by cgalvan — CHANGES_REQUESTED** [review]

**Review by byrcolin — APPROVED** [review]

File: Code/Editor/Controls/ConsoleSCB.cpp (hunk 1 above)
Comment (sphrose) [reply Code/Editor/Controls/ConsoleSCB.cpp:308]: Thanks, removed.

File: Code/Editor/Controls/ConsoleSCB.cpp (hunk 2 above)
Comment (sphrose) [reply Code/Editor/Controls/ConsoleSCB.cpp:409]: Fixed.

**Review by cgalvan — APPROVED** [review]

**Pull request merged**